src/averageAges.js

```diff
@@ -54,8 +83,22 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const children = onlyWithSon
+    ? getMen(people)
+    : people;
+  const mothers = getMothers(people);
+  const ageDiffs = children.map(({ mother: motherName, born: childBorn }) => {
+    const mother = mothers.filter(({ name }) => name === motherName);
```

Comment:
```suggestion
    const mother = mothers.find(({ name }) => name === motherName);
```
If you need only first occurrence - use `find`